### PR TITLE
update for Fastify v5

### LINKED ISSUE #26: Updating for v5
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

I'm looking to update this repo for v5. https://github.com/fastify/fastify/issues/5116

Changes Needed (Please let me know if anything might be missing)
- Pin tap to 16.3.9
- Update workflows to use v4 (temporarily pointing directly to a git sha)
- ~Update `engines` in package.json to >= 18~ Remove engines from package.json
- Update dependabot config to ignore major tap versions

### PR DESCRIPTION
**Pull request by gurgunday**
Resolves #26

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ function Negotiator (options) {
   const {
     supportedValues = [],
     cache
-  } = (options && typeof options === 'object' && options) || {}
+  } = (typeof options === 'object' && options) || {}
```
Comment (Uzlopak) [inline index.js:11]: why?

Comment (gurgunday) [reply index.js:11]: What does it do? It looks funny 😄

But I see it's irrelevant

Comment (Uzlopak) [reply index.js]: first you have the null check, then you check if is an object, if true return that object. seems odd, but it should be faster than having the string comparison first. ;)

**Review by Uzlopak — APPROVED** [review]
LGTM

File: index.js (hunk 1 above)
Comment (gurgunday) [reply index.js]: I think `typeof` no longer does string comparison, it has specific bytecode so it's as fast as it gets

Will check it again though

Comment (gurgunday) [reply index.js]: FYI:

```js
function isString(exp) {
  return typeof exp === "string";
}

console.log(isString("hello"));
```


```sh
   27 S> 0xeb02adde136 @    0 : 0b 03             Ldar a0
         0xeb02adde138 @    2 : 20 01             TestTypeOf #1
   58 S> 0xeb02adde13a @    4 : a9                Return
```

**Comment by gurgunday** [comment]
Removed the `engines` field in for consistency with other plugins and https://github.com/fastify/accept-negotiator/issues/26#issuecomment-1790633565

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**